common/utils/text_extractors.py

```diff
@@ -112,6 +113,27 @@ async def _process_file_async(self, file_path, folder_path_obj, graphname):
                 graphname
             )
 
+            # If temp_folder provided, save immediately and return metadata only
+            if temp_folder and doc_entries:
+                saved_files = []
+                for idx, doc_data in enumerate(doc_entries):
+                    # Use file_counter for unique naming across all files
+                    counter_val = next(file_counter) if file_counter else idx
+                    doc_filename = f"doc_{counter_val}_{doc_data.get('doc_id', 'unknown')}.json"
```

Comment:
Given we're already write text/image to separate json files, we can consider using the old jsonl format to include everything in a single jsonl file, which can be loaded directly with conn.runLoadingJobWithFile() without needing extra processing. @prinskumar-tigergraph